packages/core/src/contracts/parameterizer.ts

```diff
@@ -39,34 +40,240 @@ export class Parameterizer extends BaseWrapper<ParameterizerContract> {
     super(web3Wrapper, instance);
   }
 
+  /**
+   * Returns Voting instance associated with this TCR
+   */
+  public async getVoting(): Promise<Voting> {
+    return Voting.atUntrusted(this.web3Wrapper, await this.getVotingAddress());
+  }
+
+  /**
+   * Get address for voting contract used with this TCR
+   */
+  public async getVotingAddress(): Promise<EthAddress> {
+    return this.instance.voting.callAsync();
+  }
+
+  public async getPropState(propID: string): Promise<ParamProposalState> {
+    if (! await this.instance.propExists.callAsync(propID)) {
```

Comment:
might want to add parentheses around await

Reply:
Also, we can return an enum from the smart-contracts which will be converted to bignum. Might as well return the state with one call here and iterate on that.